### DISCUSSION TITLE
Killing dosemu2 in -dumb -td -kt mode within GNU screen within mate-terminal with "killall -v dosemu.bin" results in mouse input being sent as text

### DISCUSSION THREAD

**Discussion by ecm-pushbx** [dosemu2/dosemu2#2061] [Q&A]
When I run dosemu2 in `-dumb -td -kt` mode, in a mate-terminal window running GNU screen running bash, and then I terminate dosemu2 using a `killall -v dosemu.bin` command, then afterwards the terminal window that ran dosemu2 will receive mouse input events as a bunch of text it appears. I do not know how to reset things from this state, save for closing this GNU screen window and opening a new one.

The following shows the command with which I'm running dosemu2. At the end you can see some of the spam generated by a few mouse cursor movements. This problem does not happen when I'm running just `dosemu` (with its own graphical window). I believe I first came across this problem when the dosemu2 instance crashed with some fault rather than being terminated by `killall`. The problem is the same if I run dosemu2 in a new mate-terminal window without GNU screen being involved.

![Screenshot at 2023-07-31 18-31-59](https://github.com/dosemu2/dosemu2/assets/31538682/6c94cd33-fda8-4054-a649-32d899ea1830)

[boot.log](https://github.com/dosemu2/dosemu2/files/12217262/boot.log)

Any ideas on how to reset this state without exiting that window, or how to avoid it to begin with? This is not a bug report because I assume that there is no bug here, but I would like to learn how to cope with this better.

**Comment by ecm-pushbx** [comment]
Found it with my partner's help. She suggested `reset`. I'm now using `tset` to disable the mouse input but not clear the screen (as `reset` would).

**Comment by ecm-pushbx** [comment]
Hmm. `tset` doesn't always work. Need to test more.

**Comment by stsp** [comment]
There are other signals, like SIGQUIT,
that should disable mouse properly.
Dont remember right now how do we
handle SIGTERM.

**Reply by ecm-pushbx** [reply]
Thanks, that works!

**Reply by stsp** [reply]
I added the xterm recovery code, so now
even SIGTERM should work.
Please check.

**Reply by ecm-pushbx** [reply]
It does work now, both with SIGTERM and when crashing dosemu2 (eg "`mov ax, word [0FFFFh]`" in KVM Virtual 86 Mode).

**Comment by stsp** [comment]
Thanks, so SIGTERM or SIGKILL should
work properly these days. Signal-related
clean-ups are completely optional.
If you see some resource leak on SIGKILL
or an observable lack of a clean-up, this
is a bug.
